SpatialGDK/Source/Private/Interop/SpatialSender.cpp

```diff
@@ -0,0 +1,585 @@
+// Copyright (c) Improbable Worlds Ltd, All Rights Reserved
+
+#include "SpatialSender.h"
+
+#include "EngineClasses/SpatialActorChannel.h"
+#include "EngineClasses/SpatialNetBitWriter.h"
+#include "EngineClasses/SpatialNetDriver.h"
+#include "EngineClasses/SpatialPackageMapClient.h"
+#include "Interop/Connection/SpatialWorkerConnection.h"
+#include "Interop/SpatialReceiver.h"
+#include "Schema/Rotation.h"
+#include "Schema/StandardLibrary.h"
+#include "Schema/UnrealMetadata.h"
+#include "SpatialConstants.h"
+#include "Utils/ComponentFactory.h"
+#include "Utils/RepLayoutUtils.h"
+
+void USpatialSender::Init(USpatialNetDriver* InNetDriver)
+{
+	NetDriver = InNetDriver;
+	Connection = InNetDriver->Connection;
+	PackageMap = InNetDriver->PackageMap;
+	TypebindingManager = InNetDriver->TypebindingManager;
+	Receiver = InNetDriver->Receiver;
+}
+
+Worker_RequestId USpatialSender::CreateEntity(const FString& ClientWorkerId, const FString& EntityType, USpatialActorChannel* Channel)
+{
+	AActor* Actor = Channel->Actor;
+
+	WorkerAttributeSet WorkerAttribute = { TEXT("UnrealWorker") };
+	WorkerAttributeSet ClientAttribute = { TEXT("UnrealClient") };
+	WorkerAttributeSet OwningClientAttribute = { TEXT("workerId:") + ClientWorkerId };
+
+	WorkerRequirementSet WorkersOnly = { WorkerAttribute };
+	WorkerRequirementSet ClientsOnly = { ClientAttribute };
+	WorkerRequirementSet OwningClientOnly = { OwningClientAttribute };
+
+	WorkerRequirementSet AnyUnrealWorkerOrClient = { WorkerAttribute, ClientAttribute };
+	WorkerRequirementSet AnyUnrealWorkerOrOwningClient = { WorkerAttribute, OwningClientAttribute };
+
+	WorkerRequirementSet ReadAcl;
+	if (Actor->GetClass()->HasAnySpatialClassFlags(SPATIALCLASS_ServerOnly))
+	{
+		ReadAcl = WorkersOnly;
+	}
+	else if (Actor->IsA<APlayerController>())
```

Comment:
Lets add that ticket to create an additional spatial flag to track `OwningClient` flag